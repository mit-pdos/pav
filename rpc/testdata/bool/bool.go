package rpc

type arg struct {
	x bool
}
