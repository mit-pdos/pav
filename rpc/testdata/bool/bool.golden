// Auto-generated from spec "github.com/mit-pdos/pav/rpc/testdata/bool/bool.go"
// using compiler "github.com/mit-pdos/pav/rpc".
package rpc

import (
	"github.com/mit-pdos/pav/marshalutil"
	"github.com/tchajed/marshal"
)

type errorTy = bool

const (
	errNone errorTy = false
	errSome errorTy = true
)

func (o *arg) encode() []byte {
	var b = make([]byte, 0)
	b = marshalutil.WriteBool(b, o.x)
	return b
}
func (o *arg) decode(b0 []byte) ([]byte, errorTy) {
	var b = b0
	x, b, err := marshalutil.ReadBool(b)
	if err {
		return nil, err
	}
	o.x = x
	return b, errNone
}
